module/Database/src/Service/Mailman.php

```diff
@@ -210,6 +214,15 @@ private function syncMembershipSingle(
 
         $verifyTime = (new DateTime())->sub(new DateInterval('P1D'));
 
+        $isMailmanList = $dbList->hasMailmanList();
+        if ($isMailmanList) {
+            $listId = $dbList->getMailmanList()->getMailmanId();
+            $knownMembers = $this->getMailmanListSubscriberEmails($listId);
```

Comment:
Just a note, the current PHP processes are very much limited in terms of RAM usage. We already see the issues with the nightly ReportDB generation runs, so we have to keep an eye on this to see whether it slows other things down too much when a sync happens.